packages/data-context/src/data/ProjectConfigManager.ts

```diff
@@ -0,0 +1,490 @@
+      this._registeredEventsTarget = this._testingType
+      const config = await this.getFullInitialConfig()
+      const setupNodeEventsReply = await this._eventsIpc?.callSetupNodeEventsWithConfig(this._testingType, config, this.options.handlers)
+
+      await this.handleSetupTestingTypeReply(this._eventsIpc, loadConfigReply, setupNodeEventsReply)
+      this._state = 'ready'
+    } catch (error) {
+      debug(`catch setupNodeEvents %o`, error)
+      this._state = 'errored'
+      if (this._eventsIpc) {
+        this._eventsIpc.cleanupIpc()
+      }
+
+      this.closeWatchers()
+
+      throw error
+    } finally {
+      this.options.ctx.emitter.toLaunchpad()
+    }
+  }
+
+  private async handleSetupTestingTypeReply (ipc: ProjectConfigIpc, loadConfigReply: LoadConfigReply, result: SetupNodeEventsReply) {
+    this.options.eventRegistrar.reset()
+
+    for (const { event, eventId } of result.registrations) {
+      debug('register plugins process event', event, 'with id', eventId)
+
+      this.options.eventRegistrar.registerEvent(event, function (...args: any[]) {
+        return new Promise((resolve, reject) => {
+          const invocationId = _.uniqueId('inv')
+
+          debug('call event', event, 'for invocation id', invocationId)
+
+          ipc.once(`promise:fulfilled:${invocationId}`, (err: any, value: any) => {
+            if (err) {
+              debug('promise rejected for id %s %o', invocationId, ':', err.stack)
+              reject(_.extend(new Error(err.message), err))
+
+              return
+            }
+
+            if (value === UNDEFINED_SERIALIZED) {
+              value = undefined
+            }
+
+            debug(`promise resolved for id '${invocationId}' with value`, value)
+
+            return resolve(value)
+          })
+
+          const ids = { invocationId, eventId }
+
+          // no argument is passed for cy.task()
+          // This is necessary because undefined becomes null when it is sent through ipc.
+          if (event === 'task' && args[1] === undefined) {
+            args[1] = {
+              __cypress_task_no_argument__: true,
+            }
+          }
+
+          ipc.send('execute:plugins', event, ids, args)
+        })
+      })
+    }
+
+    const cypressEnv = await this.loadCypressEnvFile()
+    const fullConfig = await this.buildBaseFullConfig(loadConfigReply.initialConfig, cypressEnv, this.options.ctx.modeOptions)
+
+    const finalConfig = this._cachedFullConfig = this.options.ctx._apis.configApi.updateWithPluginValues(fullConfig, result.setupConfig ?? {})
+
+    await this.options.onFinalConfigLoaded(finalConfig)
+
+    this.watchFiles([
+      ...loadConfigReply.requires,
+      ...result.requires,
+      this.configFilePath,
+      this.envFilePath,
+    ])
+
+    return result
+  }
+
+  async reloadConfig () {
+    this._loadConfigPromise = undefined
+    this._registeredEventsTarget = undefined
+
+    await this.initializeConfig()
+    this.loadTestingType()
```

Comment:
This should be awaited since it has only promise returns
```suggestion
    await this.loadTestingType()
```